src/components/sidebar/SafeList/index.tsx

```diff
@@ -75,27 +79,59 @@ const SafeList = ({ closeDrawer }: { closeDrawer?: () => void }): ReactElement =
   const hasNoSafes = Object.keys(ownedSafes).length === 0 && Object.keys(addedSafes).length === 0
   const isWelcomePage = router.pathname === AppRoutes.welcome
 
+  const handleFetchQueued = async () => {
+    const fetchedQueuedTxs: Record<string, string | undefined> = {}
+
+    for (let [chainId, safes] of Object.entries(addedSafes)) {
+      for (let safeAddress of Object.keys(safes)) {
+        const result = await getTransactionQueue(chainId, safeAddress)
+
+        if (result.results.length === 0) continue
+
+        fetchedQueuedTxs[safeAddress] = `${result.results.filter(isTransactionListItem).length}${
+          result.next ? '+' : ''
+        }`
+      }
+    }
+
+    setSafeQueuedTxs(fetchedQueuedTxs)
+  }
+
   return (
     <div className={css.container}>
-      <div className={css.header}>
-        <Typography variant="h4" display="inline" fontWeight={700}>
-          My Safes
-        </Typography>
-        {!isWelcomePage && (
-          <Track {...OVERVIEW_EVENTS.ADD_SAFE}>
-            <Link href={{ pathname: AppRoutes.welcome }} passHref>
-              <Button
-                disableElevation
-                size="small"
-                variant="outlined"
-                onClick={closeDrawer}
-                startIcon={<SvgIcon component={AddIcon} inheritViewBox fontSize="small" />}
-              >
-                Add
-              </Button>
-            </Link>
-          </Track>
-        )}
+      <div className={css.headerWrapper}>
+        <div className={css.header}>
+          <Typography variant="h4" display="inline" fontWeight={700}>
+            My Safes
+          </Typography>
+          <div className={css.headerButtons}>
+            <Button
+              disableElevation
+              size="small"
+              variant="outlined"
+              onClick={handleFetchQueued}
+              startIcon={<SvgIcon component={UpdateIcon} inheritViewBox fontSize="small" />}
+              sx={{ color: ({ palette }) => palette.info.main }}
```

Comment:
The blue outlined button is hard to read, especially in light mode. Not sure if its only an issue on my screen but running it through a contrast checker also fails: https://webaim.org/resources/contrastchecker/?fcolor=5FDDFF&bcolor=FFFFFF

Reply:
cc @liliiaorlenko

Reply:
I've removed the button. Now the queues are requested when opening the Safe list sidebar. I've updated the PR description.